.github/workflows/unit-test.yml

```diff
@@ -11,26 +11,16 @@ jobs:
     name: Unit tests
     runs-on: ubuntu-latest
     steps:
-
-      - name: Setup Go 1.19.4
-        uses: actions/setup-go@v1
+      - name: Setup Go 1.19
+        uses: actions/setup-go@v3
         with:
-          go-version: 1.19.4
+          go-version: 1.19
 
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
-      - name: Cache Go build and dependencies
-        uses: actions/cache@v3
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          # make sure concurrent runs (not really supported) do not match the same key but instead fall back to a reasonable cache
-          key: ${{ runner.os }}-go-${{ hashFiles('go.sum') }}-${{ github.event.action }}${{ github.event.after }}
-          restore-keys: |
-            ${{ runner.os }}-go-${{ hashFiles('go.sum') }}
-            ${{ runner.os }}-go-
```

Comment:
The go build cache should still be useful. Because we were probably [trashing](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy) all of our cache space on Github Actions with all the Docker stuff this wasn't effective. But if we get rid of this, it should be fine and speed up the unit tests